share/models/change.py

```diff
@@ -89,7 +93,7 @@ def accept(self, save=True):
         ret = []
         with transaction.atomic():
             self._changes_cache = list(self.changes.all())
-            for c in self._changes_cache:
+            for c in sorted(self._changes_cache, key=lambda c: c.type):
```

Comment:
`sorted` is a stable sort, so this shouldn't be a problem. But now that you mention it, it would be better to but merge nodes first while sorting the ChangeGraph, so the order is always consistent.